File: server/app/schemas/trade.py
"""
交易模式
"""

from pydantic import BaseModel
from typing import Optional, Literal
from datetime import datetime


class OrderCreate(BaseModel):
    """创建订单"""
    stock_code: str
    stock_name: Optional[str] = None
    type: Literal["BUY", "SELL"]
    order_type: Literal["MARKET", "LIMIT"]
    price: Optional[float] = None
    quantity: int


class Order(BaseModel):
    """订单"""
    id: str
    stock_code: str
    stock_name: str
    type: Literal["BUY", "SELL"]
    order_type: Literal["MARKET", "LIMIT"]
    price: Optional[float] = None
    quantity: int
    status: Literal["PENDING", "FILLED", "CANCELLED", "REJECTED"]
    filled_quantity: int
    filled_price: float
    # 费用明细
    stamp_tax: float = 0.0
    commission: float = 0.0
    transfer_fee: float = 0.0
    total_fee: float = 0.0
    slippage: float = 0.0
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class Position(BaseModel):
    """持仓"""
    stock_code: str
    stock_name: str
    quantity: int
    cost_price: float
    current_price: float
    market_value: float
    profit: float
    profit_percent: float
    total_fees: float = 0.0
    realized_profit: float = 0.0


class AccountInfo(BaseModel):
    """账户信息"""
    total_asset: float
    available_cash: float
    market_value: float
    profit: float
    profit_percent: float
    total_fees_paid: float = 0.0

File: extension/src/types/trade.ts
/**
 * 交易类型定义
 */

import { BaseEntity } from './common';

export type OrderType = 'BUY' | 'SELL';
export type OrderStatus = 'PENDING' | 'FILLED' | 'CANCELLED' | 'REJECTED';
export type PriceType = 'MARKET' | 'LIMIT';

export interface Order extends BaseEntity {
    stockCode: string;
    stockName: string;
    type: OrderType;
    orderType: PriceType;
    price?: number; // 限价单价格
    quantity: number;
    status: OrderStatus;
    filledQuantity: number;
    filledPrice: number;
}

export interface Position {
    stockCode: string;
    stockName: string;
    quantity: number;
    costPrice: number;
    currentPrice: number;
    marketValue: number;
    profit: number;
    profitPercent: number;
}

export interface AccountInfo {
    totalAsset: number;
    availableCash: number;
    marketValue: number;
    profit: number;
    profitPercent: number;
}

